### PR TITLE
vagrant: use a matching version of Weave Net

### PR DESCRIPTION
**Pull request by nixpanic**
While deploying the Vagrant environment, Kubernetes 1.9.2 is currently
used. This version does not seem to be compatible with the 1.6 version
of Weave Net. While the weave- pods are starting, the following errors
are reported:

  dockerd-current[6461]: ERROR: logging before flag.Parse: E0119 15:50:15.165210    9894 reflector.go:205] github.com/weaveworks/weave/prog/weave-npc/main.go:230: Failed to list *v1.NetworkPolicy: networkpolicies.networking.k8s.io is forbidden: User "system:serviceaccount:kube-system:weave-net" cannot list networkpolicies.networking.k8s.io at the cluster scope

By using the recommended way to install the add-on, a matching version
will automatically be picked.

URL: https://www.weave.works/docs/net/latest/kubernetes/kube-addon/
Suggested-by: John Mulligan <jmulligan@redhat.com>
Signed-off-by: Niels de Vos <ndevos@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/421)

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
There's a version number we're getting from the packages in the common tasks:
https://github.com/gluster/gluster-kubernetes/blob/master/vagrant/roles/common/tasks/main.yml#L93

Could we possibly re-use this variable?

Also I didn't think the "command" task supported subshells, I thought you needed the "shell" for that.

**Comment by nixpanic** [comment]
No, the `kube_ver` variable can not be used, the web service that returns the weave-net yaml file expects a base64 encoded json string.

For the `command` vs `shell` bit, I thought I tested this and things worked. Let me try again and report back.

**Comment by phlogistonjohn** [comment]
Ok, even if it works, it feels more "ansible-y" to run that part in a separate command, capture that result, and then use it in the latter command.

**Comment by jarrpa** [comment]
@nixpanic To @phlogistonjohn's point, I have had the following pending change in my working branch for a while and it seems to work:
```
-  command: kubectl apply -f https://git.io/weave-kube-1.6
+  command: "kubectl apply -f https://cloud.weave.works/k8s/v{{ kube_ver.stdout[:3] }}/net"
```
Is something like this not quite right?

**Comment by nixpanic** [comment]
@jarrpa, I could not find any reference to `kubectl apply -f https://cloud.weave.works/k8s/v{{ kube_ver.stdout[:3] }}/net` on [this page](https://www.weave.works/docs/net/latest/kubernetes/kube-addon/). If that is a supported way of installing weave, it seems ok (until kubernetes 1.10 is released?!).

**Comment by nixpanic** [comment]
My local testing has finished, and it seems to work well.

**Comment by jarrpa** [comment]
My testing shows this:
```
TASK [master : create weave network] *******************************************
fatal: [master]: FAILED! => {"changed": true, "cmd": ["kubectl", "apply", "-f", "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\\n')"], "delta": "0:00:00.852793", "end": "2018-01-24 13:51:40.802537", "msg": "non-zero return code", "rc": 1, "start": "2018-01-24 13:51:39.949744", "stderr": "error: unable to read URL \"https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\\\\n')\", server reported 400 Bad Request, status code=400", "stderr_lines": ["error: unable to read URL \"https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\\\\n')\", server reported 400 Bad Request, status code=400"], "stdout": "", "stdout_lines": []}
```
Using `ansible-2.4.2.0-1.fc27.noarch`. Looks like the string isn't getting parsed?

**Comment by nixpanic** [comment]
@jarrpa, I just updated the patch again, with the changes John suggested. You seem to have tried the previous version.

**Comment by jarrpa** [comment]
Been testing since you commented. The WeaveNet step seems to be working, I just want to make sure if a failure I'm seeing afterwards is my fault or somehow related.

**Comment by jarrpa** [comment]
Yup, definitely my fault. :P LGTM!
